pilot/pkg/bootstrap/options.go

```diff
@@ -136,8 +136,23 @@ var serviceAccountVar = env.RegisterStringVar("SERVICE_ACCOUNT", "", "")
 
 var revisionVar = env.RegisterStringVar("REVISION", "", "")
 
+// NewPilotArgs constructs pilotArgs with default values.
+func NewPilotArgs(initFuncs ...func(*PilotArgs)) *PilotArgs {
+	p := &PilotArgs{}
+
+	// Apply custom initialization functions.
+	for _, fn := range initFuncs {
+		fn(p)
+	}
+
+	// Apply Default Values.
+	p.applyDefaults()
```

Comment:
This is confusing, the order should be : 1. default value, 2. custom value

Reply:
Yes. But tests are failing if we flip because the the applyDefaults sets some values only if custom init does not set. I will see what is that dependency